library/src/main/java/com/github/lukaspili/reactivebilling/observable/BaseObservable.java

```diff
@@ -38,13 +38,19 @@ public void call(Subscriber<? super T> subscriber) {
         final Connection connection = new Connection(subscriber, useSemaphore);
 
         ReactiveBilling.log(null, "Bind service (thread %s)", Thread.currentThread().getName());
+        boolean bound = false;
         try {
-            context.bindService(intent, connection, Context.BIND_AUTO_CREATE);
+            bound = context.bindService(intent, connection, Context.BIND_AUTO_CREATE);
         } catch (SecurityException e) {
             ReactiveBilling.log(e, "Bind service error");
             subscriber.onError(e);
         }
 
+        //service is not bound, throw exception
+        if(!bound){
+            subscriber.onError(new IllegalStateException("Service unable to bind"));
```

Comment:
If we can't bind to the service, do we need to exit here as opposed to continuing below?